### ISSUE TITLE
Tiered storage migration is slow and fails frequently

### ISSUE THREAD

**Issue by divanikus** [chrislusf/seaweedfs#1991]
**Describe the bug**
I've been using tiered migration feature since it was added in 2.28, but after updating to 2.35 and later to 2.39 my setup experience problems with migration. It is usually really slow now and often ends up with failure.

I'm using 5GB volumes with 010 replication. Volumes are copied approximately 5 minutes per volume (which is too slow for a 1Gbps network) and often ends up like this:

```
master: localhost:9333 filer: localhost:8888
I0413 15:05:19 15470 masterclient.go:126] redirected to leader 192.168.65.54:9333
> > collect ssd volumes quiet for: 1 seconds
tier move volumes: [6326 6299 6323 6348 6329 6328 6304 6302 6298 6317]
moving volume 6326 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6326 on 192.168.65.63:8080 ...
markVolumeReadonly 6326 on 192.168.65.62:8080 ...
2021/04/13 15:05:19 copying volume 6326 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6326: move volume 6326 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6326 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unknown desc = failed to copy /var/lib/weed/volume/offline-photo-2021-04-07_6326.dat file: receiving /var/lib/weed/volume/offline-photo-2021-04-07_6326.dat: rpc error: code = Unavailable desc = transport is closing
moving volume 6299 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6299 on 192.168.65.63:8080 ...
markVolumeReadonly 6299 on 192.168.65.62:8080 ...
2021/04/13 15:11:06 copying volume 6299 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6299: move volume 6299 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6299 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unknown desc = failed to copy /var/lib/weed/volume/offline-photo-2021-04-07_6299.dat file: receiving /var/lib/weed/volume/offline-photo-2021-04-07_6299.dat: rpc error: code = Unavailable desc = transport is closing
moving volume 6323 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6323 on 192.168.65.63:8080 ...
markVolumeReadonly 6323 on 192.168.65.62:8080 ...
2021/04/13 15:16:32 copying volume 6323 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6323: move volume 6323 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6323 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unavailable desc = transport is closing
moving volume 6348 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6348 on 192.168.65.63:8080 ...
markVolumeReadonly 6348 on 192.168.65.62:8080 ...
2021/04/13 15:22:41 copying volume 6348 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6348: move volume 6348 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6348 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unavailable desc = transport is closing
moving volume 6329 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6329 on 192.168.65.63:8080 ...
markVolumeReadonly 6329 on 192.168.65.62:8080 ...
2021/04/13 15:28:24 copying volume 6329 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6329: move volume 6329 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6329 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unavailable desc = transport is closing
moving volume 6328 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6328 on 192.168.65.63:8080 ...
markVolumeReadonly 6328 on 192.168.65.62:8080 ...
2021/04/13 15:33:51 copying volume 6328 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6328: move volume 6328 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6328 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unavailable desc = timed out waiting for server handshake
moving volume 6304 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6304 on 192.168.65.63:8080 ...
markVolumeReadonly 6304 on 192.168.65.62:8080 ...
2021/04/13 15:34:11 copying volume 6304 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6304: move volume 6304 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6304 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unavailable desc = timed out waiting for server handshake
moving volume 6302 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6302 on 192.168.65.63:8080 ...
markVolumeReadonly 6302 on 192.168.65.62:8080 ...
2021/04/13 15:34:11 copying volume 6302 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6302: move volume 6302 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6302 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unavailable desc = timed out waiting for server handshake
moving volume 6298 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6298 on 192.168.65.63:8080 ...
markVolumeReadonly 6298 on 192.168.65.62:8080 ...
2021/04/13 15:34:11 copying volume 6298 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6298: move volume 6298 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6298 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unavailable desc = timed out waiting for server handshake
moving volume 6317 from 192.168.65.62:8080 to 192.168.65.58:8080 with disk type hdd ...
markVolumeReadonly 6317 on 192.168.65.63:8080 ...
markVolumeReadonly 6317 on 192.168.65.62:8080 ...
2021/04/13 15:34:11 copying volume 6317 from 192.168.65.62:8080 to 192.168.65.58:8080
tier move volume 6317: move volume 6317 192.168.65.63:8080 => 192.168.65.58:8080 : copy volume 6317 from 192.168.65.62:8080 to 192.168.65.58:8080: rpc error: code = Unavailable desc = timed out waiting for server handshake
```
`192.168.65.62 ` and `192.168.65.63` are NVMe SSD servers, `192.168.65.58` has plane ol' HDDs.

On `192.168.65.58` logs aren't very helpful:
```
Apr 13 15:11:06 wv-offline-02 weed[13833]: I0413 15:11:06 13833 volume_grpc_client_to_master.go:208] Volume Server Failed to talk with master 192.168.65.54:9333: EOF
Apr 13 15:11:06 wv-offline-02 weed[13833]: I0413 15:11:06 13833 volume_grpc_client_to_master.go:76] heartbeat error: EOF
Apr 13 15:11:13 wv-offline-02 weed[13833]: I0413 15:11:13 13833 volume_grpc_client_to_master.go:114] Heartbeat to: 192.168.65.53:9333
Apr 13 15:11:17 wv-offline-02 weed[13833]: I0413 15:11:15 13833 volume_grpc_client_to_master.go:135] Volume Server found a new master newLeader: 192.168.65.54:9333 instead of 192.168.65.53:9333
Apr 13 15:11:28 wv-offline-02 weed[13833]: I0413 15:11:20 13833 volume_grpc_client_to_master.go:114] Heartbeat to: 192.168.65.54:9333
Apr 13 15:16:33 wv-offline-02 weed[13833]: I0413 15:16:33 13833 volume_grpc_client_to_master.go:208] Volume Server Failed to talk with master 192.168.65.54:9333: EOF
Apr 13 15:16:33 wv-offline-02 weed[13833]: I0413 15:16:33 13833 volume_grpc_client_to_master.go:76] heartbeat error: EOF
Apr 13 15:16:39 wv-offline-02 weed[13833]: I0413 15:16:39 13833 volume_grpc_client_to_master.go:114] Heartbeat to: 192.168.65.54:9333
Apr 13 15:23:18 wv-offline-02 weed[13833]: I0413 15:23:17 13833 volume_grpc_client_to_master.go:208] Volume Server Failed to talk with master 192.168.65.54:9333: EOF
Apr 13 15:23:18 wv-offline-02 weed[13833]: I0413 15:23:18 13833 volume_grpc_client_to_master.go:76] heartbeat error: EOF
Apr 13 15:23:27 wv-offline-02 weed[13833]: I0413 15:23:27 13833 volume_grpc_client_to_master.go:114] Heartbeat to: 192.168.65.53:9333
Apr 13 15:23:30 wv-offline-02 weed[13833]: I0413 15:23:30 13833 volume_grpc_client_to_master.go:135] Volume Server found a new master newLeader: 192.168.65.54:9333 instead of 192.168.65.53:9333
Apr 13 15:23:34 wv-offline-02 weed[13833]: I0413 15:23:34 13833 volume_grpc_client_to_master.go:114] Heartbeat to: 192.168.65.54:9333
Apr 13 15:28:38 wv-offline-02 weed[13833]: I0413 15:28:38 13833 volume_grpc_client_to_master.go:208] Volume Server Failed to talk with master 192.168.65.54:9333: EOF
Apr 13 15:28:38 wv-offline-02 weed[13833]: I0413 15:28:38 13833 volume_grpc_client_to_master.go:76] heartbeat error: EOF
Apr 13 15:28:44 wv-offline-02 weed[13833]: I0413 15:28:44 13833 volume_grpc_client_to_master.go:114] Heartbeat to: 192.168.65.54:9333
Apr 13 15:34:30 wv-offline-02 weed[13833]: I0413 15:34:30 13833 volume_grpc_client_to_master.go:208] Volume Server Failed to talk with master 192.168.65.54:9333: EOF
Apr 13 15:34:30 wv-offline-02 weed[13833]: I0413 15:34:30 13833 volume_grpc_client_to_master.go:76] heartbeat error: EOF
Apr 13 15:34:36 wv-offline-02 weed[13833]: I0413 15:34:36 13833 volume_grpc_client_to_master.go:114] Heartbeat to: 192.168.65.53:9333
Apr 13 15:34:37 wv-offline-02 weed[13833]: I0413 15:34:37 13833 volume_grpc_client_to_master.go:135] Volume Server found a new master newLeader: 192.168.65.54:9333 instead of 192.168.65.53:9333
Apr 13 15:34:40 wv-offline-02 weed[13833]: I0413 15:34:40 13833 volume_grpc_client_to_master.go:114] Heartbeat to: 192.168.65.54:9333
```
On the current leader master it looks like the destination volume server is considered dead and some point and got readded.

```
Apr 13 15:10:55 wm-offline-01 weed[32384]: I0413 15:10:55 32384 topology_event_handling.go:65] Removing Volume 2157 from the dead volume server 192.168.65.58:8080
Apr 13 15:10:55 wm-offline-01 weed[32384]: I0413 15:10:55 32384 volume_layout.go:369] Volume 2157 has 1 replica, less than required 2
Apr 13 15:10:55 wm-offline-01 weed[32384]: I0413 15:10:55 32384 volume_layout.go:343] Volume 2157 becomes unwritable
Apr 13 15:10:55 wm-offline-01 weed[32384]: I0413 15:10:55 32384 topology_event_handling.go:65] Removing Volume 4979 from the dead volume server 192.168.65.58:8080
Apr 13 15:10:55 wm-offline-01 weed[32384]: I0413 15:10:55 32384 volume_layout.go:369] Volume 4979 has 1 replica, less than required 2
Apr 13 15:10:55 wm-offline-01 weed[32384]: I0413 15:10:55 32384 node.go:234] topo:dc1:sx001 removes 192.168.65.58:8080
Apr 13 15:10:55 wm-offline-01 weed[32384]: I0413 15:10:55 32384 master_grpc_server.go:29] unregister disconnected volume server 192.168.65.58:8080
Apr 13 15:11:33 wm-offline-01 weed[32384]: I0413 15:11:33 32384 node.go:222] topo:dc1:sx001 adds child 192.168.65.58:8080
Apr 13 15:11:33 wm-offline-01 weed[32384]: I0413 15:11:33 32384 node.go:222] topo:dc1:sx001:192.168.65.58:8080 adds child
Apr 13 15:11:33 wm-offline-01 weed[32384]: I0413 15:11:33 32384 master_grpc_server.go:71] added volume server 192.168.65.58:8080
Apr 13 15:11:33 wm-offline-01 weed[32384]: I0413 15:11:33 32384 volume_layout.go:355] Volume 2367 becomes writable
Apr 13 15:11:33 wm-offline-01 weed[32384]: I0413 15:11:33 32384 volume_layout.go:355] Volume 4718 becomes writable
Apr 13 15:11:33 wm-offline-01 weed[32384]: I0413 15:11:33 32384 volume_layout.go:355] Volume 1772 becomes writable
Apr 13 15:11:33 wm-offline-01 weed[32384]: I0413 15:11:33 32384 volume_layout.go:355] Volume 637 becomes writable
Apr 13 15:11:33 wm-offline-01 weed[32384]: I0413 15:11:33 32384 volume_layout.go:355] Volume 4609 becomes writable
Apr 13 15:11:33 wm-offline-01 weed[32384]: I0413 15:11:33 32384 volume_layout.go:355] Volume 108 becomes writable
```

Also I've noticed that `volume.fix.replication` is also MUCH slower than it was and also often ends with failure. Everything was much faster and did work on 2.28. So it must be some kind of new option I've missed or something like that?

**System Setup**
- OS version: Debian 10.9
- output of `weed version`: version 30GB 2.39 742ab1e linux amd64
- if using filer, show the content of `filer.toml`:
```
[leveldb2]
# local on disk, mostly for simple single-machine setup, fairly scalable
# faster than previous leveldb, recommended.
enabled = true
dir = "/var/lib/weed/filer"                                       # directory to store level db files
```

**Comment by chrislusf** [comment]
There seems no change related to this feature. Possible to do a binary search to locate the first version with the slowness?

`wv-offline-02` seems disconnecting from the masters about every 5 minutes. Was it happening only when volume is moving? How was the resource usage?

**Comment by divanikus** [comment]
I could try previous versions, if they are compatible between each other. Probably something happened between 2.28 and 2.35.

>  Was it happening only when volume is moving? How was the resource usage?
Yup, otherwise everything looks stable. `load average: 5.23, 4.83, 3.06`, not a huge deal for an 8 core server...

**Comment by chrislusf** [comment]
should be safe to change to any version.

**Comment by divanikus** [comment]
So, I have played around with different versions. Looks like seaweedfs by itself wasn't responsible for slowdowns. The main reason appears to be swapping on volume server, where swap was on the same drive as the data. I have disabled swap completely and now replication and migrations are blazing fast again. But sometimes cause weed to OOM on some volume servers.

However, while playing with versions and migrations I was faced [a nasty bug](https://github.com/chrislusf/seaweedfs/issues/2001) which caused me data loss.
